agent/hcp/telemetry/otel_exporter.go

```diff
@@ -0,0 +1,66 @@
+package telemetry
+
+import (
+	"context"
+
+	hcpclient "github.com/hashicorp/consul/agent/hcp/client"
+
+	"github.com/hashicorp/go-multierror"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/aggregation"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+)
+
+// OTELExporter is a custom implementation of a OTEL Metrics SDK metrics.Exporter.
+// The exporter is used by a OTEL Metrics SDK PeriodicReader to export aggregated metrics.
+// This allows us to use a custom client - HCP authenticated MetricsClient.
+type OTELExporter struct {
+	client   hcpclient.MetricsClient
+	endpoint string
+}
+
+// NewOTELExporter returns a configured OTELExporter
+func NewOTELExporter(client hcpclient.MetricsClient, endpoint string) *OTELExporter {
+	return &OTELExporter{
+		client:   client,
+		endpoint: endpoint,
+	}
+}
+
+// Temporality returns the Cumulative temporality for metrics aggregation.
+// Telemetry Gateway stores metrics in Prometheus format, so use Cummulative aggregation as default.
+func (e *OTELExporter) Temporality(_ metric.InstrumentKind) metricdata.Temporality {
+	return metricdata.CumulativeTemporality
+}
+
+// Aggregation returns the Aggregation to use for an instrument kind.
+func (e *OTELExporter) Aggregation(kind metric.InstrumentKind) aggregation.Aggregation {
+	switch kind {
+	case metric.InstrumentKindObservableGauge:
+		return aggregation.LastValue{}
+	case metric.InstrumentKindHistogram:
+		return aggregation.ExplicitBucketHistogram{
+			Boundaries: []float64{0, 5, 10, 25, 50, 75, 100, 250, 500, 750, 1000, 2500, 5000, 7500, 10000},
+			NoMinMax:   false,
+		}
+	}
+	// for metric.InstrumentKindCounter and others, default to sum.
+	return aggregation.Sum{}
+}
+
+// Export serializes and transmits metric data to a receiver.
+func (e *OTELExporter) Export(ctx context.Context, metrics metricdata.ResourceMetrics) error {
+	if len(metrics.ScopeMetrics) == 0 || len(metrics.ScopeMetrics[0].Metrics) == 0 {
```

Comment:
Chappie: sometimes naming conditions outside of the if block gives reviewers/code viewers a better understanding of the code.


```
// emtpyMetrics is occuring through the otel code due to .. 
emtpyMetrics := len(metrics.ScopeMetrics) == 0 || len(metrics.ScopeMetrics[0].Metrics) == 0

if emtpyMetrics {

...
```

Note: Ash mentioned in an sync review of this code that we don't control the call to export and the sdk is gathering metrics and calling this. In some circumstances it's sending an empty object through when there has been no change (theory right now).

Reply:
I could iterate through all ScopeMetrics and all Metrics within (since the transform is generic). That would make this code more extensible in case someone uses this with multiple Meters (so multiple ScopeMetrics). I think that would address your concern @loshz ?

Reply:
Added a new `isEmpty()` method that checks if there are any metrics to be sent

Reply:
Yep, this makes sense. I guess I'm just unsure why we use a slice if we know there is only ever 1 item? I assume this is coming directly from he SDK?

Reply:
Yeah it's unfortunately the output of the SDK after aggregating, given to the exporter 😓